File: CS 6011/Week5/Day 22/app/src/main/java/com/example/helloandroid/MainActivity.java
package com.example.helloandroid;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void handleBtn(View view){
        TextView tv = findViewById(R.id.outputinfold);
        tv.setText("The button has been clicked");
    }

}

File: CS 6018/Lab1/app/src/main/java/com/example/a6018lab1/MainActivity.kt
package com.example.a6018lab1

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.a6018lab1.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Initialize View Binding
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        // Set up button click listeners
        binding.button1.setOnClickListener {
            val intent = Intent(this, MainActivity2::class.java)
            intent.putExtra("button_text", "Button 1 clicked!")
            startActivity(intent)
        }

        binding.button2.setOnClickListener {
            val intent = Intent(this, MainActivity2::class.java)
            intent.putExtra("button_text", "Button 2 clicked!")
            startActivity(intent)
        }

        binding.button3.setOnClickListener {
            val intent = Intent(this, MainActivity2::class.java)
            intent.putExtra("button_text", "Button 3 clicked!")
            startActivity(intent)
        }

        binding.button4.setOnClickListener {
            val intent = Intent(this, MainActivity2::class.java)
            intent.putExtra("button_text", "Button 4 clicked!")
            startActivity(intent)
        }

        binding.button5.setOnClickListener {
            val intent = Intent(this, MainActivity2::class.java)
            intent.putExtra("button_text", "Button 5 clicked!")
            startActivity(intent)
        }
    }
}
